### PR TITLE
FIX-#3689: Add black and flake8 into development environment files

### LINKED ISSUE #3689: Add black and flake8 into development environment files

### PR DESCRIPTION
**Pull request by orcahmlee**
Signed-off-by: Andrew Li <orcahmlee@gmail.com>

## What do these changes do?
Add black and flake8 into requirements-dev.txt

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3689 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date 
- [ ] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by devin-petersohn — APPROVED** [review]
Thanks @orcahmlee for the PR! LGTM, I just triggered CI and will merge when it's passing!

TeamCity tests are down for now, but since this is just a requirements change it should be fine to merge without them passing. Thanks again!

**Comment by devin-petersohn** [comment]
@orcahmlee the commit message for 39d61d1 is missing the `FIX-#3689:` piece. You can do a `git commit --amend` and edit the commit message to add it, then you will need to `git push --force` since it is modifying the commit message of an existing commit. Let me know if that doesn't make sense!

**Comment by orcahmlee** [comment]
> @orcahmlee the commit message for [39d61d1](https://github.com/modin-project/modin/commit/39d61d12c6f3c24f214918b97197c88e087a45c6) is missing the `FIX-#3689:` piece. You can do a `git commit --amend` and edit the commit message to add it, then you will need to `git push --force` since it is modifying the commit message of an existing commit. Let me know if that doesn't make sense!

@devin-petersohn I saw the commitlint fail and already fixed it. Sorry about that, it is my first commit on this project.

**Comment by orcahmlee** [comment]
> To keep all environment files for development consistent we should also add `black` and `flake8` (probably, `flake8-print` and `flake8-no-implicit-concat` too) in `environment-dev.yml`, `requirements/env_omnisci.yml`, `requirements/requirements-no-engine.yml`.

@YarShev Thanks for pointing out the conda environment yaml files should be updated too, already added it.

**Comment by orcahmlee** [comment]
Sorry for the commitlint failing again. I changed the `- FIX-#3689:` piece to `FIX-#3689:`.

**Comment by orcahmlee** [comment]
> @orcahmlee, thank you for the changes! One more thing needs to be done prior the merge. Please add a release note under `Dependencies` clause and your GH nickname under `Contributors` clause in https://github.com/modin-project/modin/blob/master/docs/release_notes/release_notes-0.15.0.rst.

Thanks @YarShev , already done.

File: requirements/env_omnisci.yml [hunk 1]
```diff
@@ -19,3 +19,5 @@ dependencies:
   - xlrd
   - sqlalchemy
   - scipy
+  - black
+  - flake8
```
Comment (YarShev) [inline requirements/env_omnisci.yml:23]:
```suggestion
  - pip:
      - black
      - flake8
```

**Review by YarShev — COMMENTED** [review]
We are close to merge this PR. I noticed one more thing to correct and left a suggestion.

File: requirements/env_omnisci.yml (hunk 1 above)
Comment (orcahmlee) [reply requirements/env_omnisci.yml]: OK, already fixed it.

**Review by YarShev — APPROVED** [review]
@orcahmlee, thank you for the changes, LGTM! Will look forward to even more commits from you to Modin.

**Pull request merged**